sonic-chassisd/scripts/chassisd

```diff
@@ -329,11 +376,106 @@ class ModuleUpdater(logger.Logger):
+            #       SYSTEM_LAG_ID_TABLE and SYSTEM_LAG_ID_SET are adjusted appropriately
+
+            script = "local host = string.gsub(ARGV[1], '%-', '%%-')\n\
+        local dev = ARGV[2]\n\
+        local tables = {'SYSTEM_NEIGH*', 'SYSTEM_INTERFACE*', 'SYSTEM_LAG_MEMBER_TABLE*'}\n\
+        for i = 1, table.getn(tables) do\n\
+            local ps = tables[i] .. '|' .. host .. '|' .. dev\n\
+            local keylist = redis.call('KEYS', tables[i])\n\
+            for j,key in ipairs(keylist) do\n\
+                if string.match(key, ps) ~= nil then\n\
+                    redis.call('DEL', key)\n\
+                end\n\
+            end\n\
+        end\n\
+        local ps = 'SYSTEM_LAG_TABLE*|' .. '(' .. host .. '|' .. dev ..'.*' .. ')'\n\
+        local keylist = redis.call('KEYS', 'SYSTEM_LAG_TABLE*')\n\
+        for j,key in ipairs(keylist) do\n\
+            local lagname = string.match(key, ps)\n\
+            if lagname ~= nil then\n\
+                redis.call('DEL', key)\n\
+                local lagid = redis.call('HGET', 'SYSTEM_LAG_ID_TABLE', lagname)\n\
+                redis.call('SREM', 'SYSTEM_LAG_ID_SET', lagid)\n\
+                redis.call('HDEL', 'SYSTEM_LAG_ID_TABLE', lagname)\n\
+            end\n\
+        end\n\
+        return"
+            self.chassis_app_db_clean_sha = self.chassis_app_db_pipe.loadRedisScript(script)
+
+        # Chassis app db cleanup of all asics of the module
+
+        # Get the module key and host name from down_modules key
+        module, lc = re.split('\|', module_host)
+
+        # Get number of asics in the module
+        fvs = self.hostname_table.get(module)
+        if isinstance(fvs, list) and fvs[0] is True:
+            fvs = dict(fvs[-1])
+            num_asics = int(fvs[CHASSIS_MODULE_INFO_NUM_ASICS_FIELD])
+        else:
+            num_asics = 0
+
+        for asic_id in range(0, num_asics):
+            asic = CHASSIS_ASIC+str(asic_id)
+
+            # Cleanup the chassis app db entries using lua script
+            redis_cmd = 'redis-cli -h redis_chassis.server -p 6380 -n 12 EVALSHA ' + self.chassis_app_db_clean_sha + ' 0 ' + lc + ' ' + asic
+            try:
+                subp = subprocess.Popen(redis_cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+                subp.communicate()
+                self.log_notice("Cleaned up chassis app db entries for {}({})/{}".format(module, lc, asic))
+            except Exception:
+                self.log_error("Failed to clean up chassis app db entries for {}({})/{}".format(module, lc, asic))
+
+
+    def module_down_chassis_db_cleanup(self):
+        if self._is_supervisor() == False:
+            return
+        time_now = time.time()
+        for module in self.down_modules:
+            if self.down_modules[module]['cleaned'] == False:
+                down_time = self.down_modules[module]['down_time']
+                delta = (time_now - down_time) / 60
```

Comment:
IMO, SYSTEM_NEIGH table requires more frequent clean up then 30mins. When the entries for a asic that is no more existing, then leaving those entries for 30mins in the neighbor table will be cause a traffic blackhole for remote neighbors of other asics. So is it possible to have a more granular 'cleanup' instead of clearing all the tables after 30mins ?